plugins/single_plugins/wrot.cpp

```diff
@@ -15,14 +22,159 @@ static double vlen(double x1, double y1) // length of vector centered at the ori
     return std::sqrt(x1 * x1 + y1 * y1);
 }
 
+enum class mode
+{
+    NONE,
+    ROT_2D,
+    ROT_3D,
+};
+
 class wf_wrot : public wf::plugin_interface_t
 {
     wf::button_callback call;
     wf::option_wrapper_t<double> reset_radius{"wrot/reset_radius"};
+    wf::option_wrapper_t<int> sensitivity{"wrot/sensitivity"};
+    wf::option_wrapper_t<bool> invert{"wrot/invert"};
+
+    std::set<wayfire_view> transformed_views;
 
     wf::pointf_t last_position;
     wayfire_view current_view;
 
+    mode current_mode;
+
+    void reset_all()
+    {
+        for (auto v : transformed_views)
+        {
```

Comment:
I have an idea. Instead of storing the views in a separate std::set and having to check whether the view is detached, why not just iterate over all views on the output and reset them?

Reply:
That makes sense, I've changed the logic to this.